framework/src/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1370,20 +1385,6 @@ private void stubDebug(String warning) {
     }
 
     /**
-     * Throw an error if the message contains line separator characters. (Any characters after a
-     * line separator are silently discarded and not shown to the user.)
-     */
-    private void ensureSingleLine(String message, String methodName) {
```

Comment:
I can't reproduce this issue. I ran the test cases manually and correctly see multi-line warnings.
Some existing uses of stub messages contained newlines, which would have resulted in this rather bizarre error.
I removed all this code. Please let me know if you know why this should be kept.

Reply:
@wmdietl This was definitely a problem and I believe the code should be retained.

Maybe it's Maven that only prints the first line (Maven does some other violence to error messages as well).  If so, the comment, when reinstated, should be more specific about what tool is at fault.  (It should have been in the first place; sorry about that.)

Reply:
@mernst Why would that only be a problem here? We don't enforce this for any of the other warnings or errors that we issue. The warnings from the parsing errors will be much easier to read when they contain newlines.

Reply:
It might be a problem elsewhere as well.  I encountered it here.  In the future we could be consistent throughout.  Or, if the problem is related to Maven, we could convince the Maven maintainers to output the whole message instead of just part.

Reply:
I have verified that the multi-line StubParser messages, both at WARNING and at NOTE level, are correctly output in the Maven build.
This PR should be merged. If another tool has problems with error messages this should be fixed for all output, not just the stubparser.